src/backend/app/users/user_routes.py

```
@@ -56,6 +56,17 @@ async def get_user_roles(current_user: Annotated[DbUser, Depends(mapper)]):
     return user_roles
 
 
+@router.get("/change-role", response_model=user_schemas.UserOut)
+async def change_user_role(
+    user: Annotated[DbUser, Depends(get_user)],
```

This seems like an extra db call for no reason - should it be `user_id: int` instead?

---

Using the dependency would raise an error if the user doesn't exist for the entered user id.

---

- Use `user_id` in path `{user_id}/change-role`
- Instead of using UserUpdate (which you have created, which is fine; you can leave it as it), just to update roles, you can use the already existing `UserRole`: `user_schemas.UserRole(role=new_role)` like this.
So that there won't be necessary db calls, the code will look clean and optimised.

---

If the user doesn't exist, you can raise an error from db update?

---

ok